ci/Jenkinsfile.test-e2e.android

```diff
@@ -0,0 +1,277 @@
+#!/usr/bin/env groovy
+library 'status-jenkins-lib@v1.9.27'
+
+/* Options section can't access functions in objects. */
+def isPRBuild = utils.isPRBuild()
+def isNightlyBuild = utils.isNightlyBuild()
+
+pipeline {
+  agent {
+    dockerfile {
+      label 'linuxcontainer'
+      filename 'tests.Dockerfile'
+      dir 'ci'
+      args '--user jenkins'
+    }
+  }
+
+  parameters {
+    gitParameter(
+      name: 'GIT_REF',
+      description: 'Git branch to checkout.',
+      branchFilter: 'origin/(.*)',
+      branch: '',
+      defaultValue: 'master',
+      quickFilterEnabled: false,
+      selectedValue: 'DEFAULT',
+      sortMode: 'ASCENDING_SMART',
+      tagFilter: '*',
+      type: 'PT_BRANCH'
+    )
+    string(
+      name: 'BUILD_SOURCE',
+      description: 'URL to APK or Jenkins build (pkg/*.apk). Required when BROWSERSTACK_APP_ID is empty.',
+      defaultValue: ''
+    )
+    string(
+      name: 'BROWSERSTACK_APP_ID',
+      description: 'Existing BrowserStack app identifier (lt://...). Leave empty to upload from BUILD_SOURCE.',
+      defaultValue: ''
+    )
+    string(
+      name: 'DEVICE_ID',
+      description: 'BrowserStack device id to run on (leave empty to use the environment default).',
+      defaultValue: ''
+    )
+    string(
+      name: 'PYTEST_ARGS',
+      description: 'Pytest flags (e.g. "-m smoke" or "-m critical") and other args.',
+      defaultValue: '-n=5 -m smoke tests'
+    )
+  }
+
+  options {
+    timestamps()
+    timeout(time: 120, unit: 'MINUTES')
+    buildDiscarder(logRotator(
+      daysToKeepStr: '30',
+      numToKeepStr: '30',
+      artifactNumToKeepStr: '30',
+    ))
+    disableRestartFromStage()
+  }
+
+  environment {
+    VIRTUAL_ENV = "${env.WORKSPACE_TMP}/venv-appium"
+    PYTHONUNBUFFERED = "1"
+    BUILD_SOURCE = "${params.BUILD_SOURCE}"
+    BROWSERSTACK_APP_ID_PARAM = "${params.BROWSERSTACK_APP_ID}"
+    PYTEST_ARGS = "${params.PYTEST_ARGS}"
+    DEVICE_ID = "${params.DEVICE_ID}"
+  }
+
+  stages {
+    stage('Checkout Reference') {
+      steps {
+        script {
+          def branch = params.GIT_REF ?: 'origin/master'
+          checkout([
+            $class: 'GitSCM',
+            branches: [[name: branch]],
+            doGenerateSubmoduleConfigurations: false,
+            extensions: scm.extensions,
+            submoduleCfg: [],
+            userRemoteConfigs: scm.userRemoteConfigs
+          ])
+          env.GIT_COMMIT = sh(script: 'git rev-parse HEAD', returnStdout: true).trim()
+          echo "Checked out ${branch} @ ${env.GIT_COMMIT}"
```

Comment:
Why are we explicitly checking out a branch? This should be done by the Jenkins Job before it even starts parsing the Jenkinsfile.